components/public-api/gitpod/experimental/v1/teams.proto

```diff
@@ -59,3 +74,53 @@ message CreateTeamRequest {
 message CreateTeamResponse {
     Team team = 1;
 }
+
+message ListTeamsRequest {
+    // TODO: pagination options
+}
+
+message ListTeamsResponse {
+    repeated Team teams = 1;
+}
+
+message JoinTeamRequest {
+    // invitation_id is the invitation ID for a Team
+    string invitation_id = 1;
+}
+
+message JoinTeamResponse {
+    // team is the team the user has just joined
+    Team team = 1;
+}
+
+message ResetTeamInvitationRequest {
+    string team_id = 1;
+}
+
+message ResetTeamInvitationResponse {
+    // team_invitation is the new invitation for the team.
+    TeamInvitation team_invitation = 1;
+}
+
+message UpdateTeamMemberRequest {
+    // team_id is the ID of the team in which the role is to be updated
+    string team_id = 1;
+
+    // team_member is the team member being updated.
+    TeamMember team_member = 2;
+
+}
+
+message UpdateTeamMemberResponse {
+    TeamMember team_member = 2;
+}
+
+message DeleteTeamMemberRequest {
+    // team_id is the ID of the team in which a member should be deleted.
+    string team_id = 1;
+
+    // team_member_id is the ID of the TeamMember that should be deleted from the team.
+    string team_member_id = 2;
+}
+
+message DeleteTeamMemberResponse {}
```

Comment:
Does this need indication of success/failure?

Reply:
It's indicated through the notion of error resulting from the call (status code). It succeeds if there's no error.
We may want to add more details about the delete here but so far I've not found the use-case and server also returns nil.

Reply:
In other words, the caller does:
```
response, err := client.DeleteTeamMember(...)
if err != nil {
  ... 
}
// got success
```